module/VuFind/src/VuFind/View/Helper/Root/Citation.php

```diff
@@ -223,56 +262,71 @@ public function getDataCSL()
         default:
             $item['type'] = 'book';
         }
-        // subtitle -> shortTile
-        if (!empty($this->details['subtitle'])) {
-            $item['shortTitle'] = $this->details['subtitle'];
-        }
-        // authors
-        if (!empty($this->details['authors'])) {
-            foreach ($this->details['authors'] as $author) {
-                $item['author'][] = ['literal' => $author];
+
+        // title
+        $this->addIfNotEmpty(
+            $item,
+            [
+                'title' => $this->driver->tryMethod('getTitle'),
+                'title-short' => $this->driver->tryMethod('getShortTitle'),
+            ]
+        );
+
+        // meta
+        $this->addIfNotEmpty(
+            $item,
+            [
+                'call-number' => $this->driver->getCallNumbers(),
+                'edition' => $this->details['edition'],
+                'ISBN' => $this->driver->getISBNs(),
+                'language' => $this->driver->getLanguages(),
+                'publisher' => $this->driver->getPublishers(),
+                'publisher-place' => $this->driver->getPlacesOfPublication(),
+            ]
+        );
+
+        // journal meta
+        $this->addIfNotEmpty(
+            $item,
+            [
+                'ISSN' => $this->driver->getISSNs(),
+                'volume' => $this->driver->getContainerIssue(),
+                'volume-title' => $this->driver->getContainerVolume(),
```

Comment:
Are you sure this is right? What if there's a container title, container volume number and container issue number, like "New York Weekly, v. 54, no. 7"? (I can't remember exactly how this is represented in the driver object, but we should double check that we're doing it right here).

Reply:
Maybe it would be worth having a quick one on one and go through the spec to make sure I'm seeing all the options. 

These are the confusing/possibly misimplemented bits:
- subtitle
- volume
- dates